src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -1,13 +1,92 @@
 import React from 'react';
-import {getFollowing} from '../../../service/profile';
+import {Context} from '../../../context';
+import DiscoveryAction from '../../../context/actions/discoveryAction';
+import following from '../../../context/actions/following';
 import {getFollowedDomain} from '../../../service/domain';
+import {getFollowing} from '../../../service/profile';
 import {getFollowingTopic} from '../../../service/topics';
-import following from '../../../context/actions/following';
-import {Context} from '../../../context';
 
 const useDiscovery = () => {
   const [, followingDispatch] = React.useContext(Context).following;
   const [refreshing, setRefreshing] = React.useState(false);
+  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
+
+  const users = React.useMemo(() => {
+    return discovery.initialUsers.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery]);
+
+  const exhangeFollower = (newUserLists, willFollow, userId) => {
+    const indexUser = newUserLists.findIndex((item) =>
+      item.user ? item.user.user_id === userId : item.user_id === userId
+    );
+    if (newUserLists[indexUser].user) {
+      newUserLists[indexUser].user.following = !!willFollow;
+    } else {
+      newUserLists[indexUser].following = !!willFollow;
+    }
+    return newUserLists[indexUser];
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `exhangeFollower` function, you are directly mutating the `newUserLists` array which is passed as an argument. This can lead to unexpected side effects if the original array was intended to be immutable. Consider creating a copy of the array before making modifications.

```diff
- newUserLists[indexUser].user.following = !!willFollow;
+ const updatedUserLists = [...newUserLists];
+ updatedUserLists[indexUser] = {
+   ...updatedUserLists[indexUser],
+   user: {
+     ...updatedUserLists[indexUser].user,
+     following: !!willFollow
+   }
+ };
+ return updatedUserLists[indexUser];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->